plugin/src/main/scala/migrate/ScalaMigratePlugin.scala

```diff
@@ -0,0 +1,101 @@
+package migrate
+
+import sbt.Keys._
+import sbt._
+import sbt.plugins.JvmPlugin
+import java.nio.file.{ Files, Path, Paths }
+
+import migrate.interfaces.Migrate
+
+import scala.collection.JavaConverters._
+import buildinfo.BuildInfo
+
+case class Scala3Inputs(scalacOptions: Seq[String], classpath: Seq[Path], classDirectory: Path)
+
+object ScalaMigratePlugin extends AutoPlugin {
+  val classpathAttribute      = AttributeKey[Seq[Path]]("unmanagedClasspath")
+  val scalacOptionAttribute   = AttributeKey[Seq[String]]("scalacOptions")
+  val classDirectoryAttribute = AttributeKey[Path]("scala3ClassDirectory")
+  val scalaBinaryVersion      = BuildInfo.scalaBinaryVersion
+  val migrateVersion          = BuildInfo.version
+  val toolClasspath           = BuildInfo.toolClasspath.split(java.io.File.pathSeparator).toList
+
+  object autoImport {
+    // val scala3CompilerOptions = taskKey[Seq[String]]("scalacOptions for scala 3")
+
+    val scala3Version     = "0.27.0-RC1"
+    val checkRequirements = taskKey[Unit]("check requirements")
+    val migrate           = taskKey[Unit]("migrate a specific project to scala 3")
+    val scala3Inputs      = taskKey[Scala3Inputs]("compute scala 3 classpath and options")
+    val storeScala3Inputs = taskKey[StateTransform]("store scala 3 classpath and options in state attributes")
+  }
+  import autoImport._
+
+  override def requires: Plugins = JvmPlugin
+
+  override def trigger = AllRequirements
+
+  override def projectSettings: Seq[Setting[_]] =
+    inConfig(Compile)(configSettings) ++
+      inConfig(Test)(configSettings)
+
+  val configSettings: Seq[Setting[_]] =
+    Def.settings(
+      migrate := migrateImp.value,
+      scala3Inputs := {
+        val state1 = Command.process(s"""set scalaVersion:="${scala3Version}" """, state.value)
+        val state2 = Command.process("storeScala3Inputs", state1)
```

Comment:
I think that those commands affect all the projects and configs but they should not:
```
        val state1 = Command.process(s"""set <project> / <config> / scalaVersion:="${scala3Version}" """, state.value)
        val state2 = Command.process("<project> / <config> / storeScala3Inputs", state1)
```
`config` can be retrieved by `configuration.value` and `project` can be retrieved by `thisProject.value` or `thisProjectRef.value`.

Reply:
I will do this change. It's interesting to be precise

Reply:
Not sure it is useful anymore since you told me that `set scalaVersion` does only change the version of the current project.

Reply:
Not sure it is useful anymore since you told me that set scalaVersion does only change the version of the current project.